src/NuGet.Core/NuGet.Common/Telemetry/Performance/ActionStepsTelemetryEvent.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Telemetry event data for nuget operation sub-steps
+    /// </summary>
+    public class ActionStepsTelemetryEvent
+    {
+        public ActionStepsTelemetryEvent(
+            string operationId,
+            string stepName,
+            string duration)
+        {
+            OperationId = operationId;
+            StepName = stepName;
+            Duration = duration;
+        }
+
+        public string OperationId { get; }
+
+        public string StepName { get; }
```

Comment:
Not really since its for our use n debugging purpose so i would like to keep it as string and have sufficient description in each step name. Also the fact that there will be different steps with different actions.